client/src/app/hole-details/hole-details.component.html

```diff
@@ -1,3 +1,7 @@
-<h3>hole-details works!</h3>
-<p>TODO: display hole details here</p>
-{{hole?.title}}
+<h3>Details</h3>
+<p>
+    <b>Par:</b> {{hole?.par}}<br>
+    <b>Start at:</b> {{hole?.startDescr}}<br>
+    <b>Finish at:</b> {{hole?.endDescr}}<br>
+    <span *ngIf="hole?.constraints != null"><b>Constraints:</b> {{hole?.constraints}}</span>
```

Comment:
Javascript has the idea of `truthy` and `falsey` values.  Because the existence of a string in JS is truthy you can change the ngIf to just be `*ngIf="hole?.constraints"` then it will evaluate as `true` if the `constraints` property exists, or `false` if it does not. .  Adding the `!= null` to it makes an explicit true or false value. 

 So while this definitely works, and does not need to change,  you can shortcut it and leave off the `!= null` part.